src/expression.c

```diff
@@ -9809,6 +9810,69 @@ Expression *SliceExp::syntaxCopy()
     return se;
 }
 
+bool SliceExp::isOpDollar(VarExp* ve)
+{
+    return ve->var == this->lengthVar;
+    // VarDeclaration* decl = ve->var->isVarDeclaration();
+    // return decl && decl->ident == Id::dollar;
+}
+
+SliceExp::Boundness SliceExp::analyzeRelativeBound(Expression* e,
+                                                   dinteger_t* p, dinteger_t* q) // out arguments
+{
+    if (e->op == TOKint64 && e->toInteger() == 0)
+    {
+        e->warning("Avoiding boundscheck for 0");
+        return atStart;
+    }
+    else if (e->op == TOKvar && this->isOpDollar((VarExp*)e)) // TODO functionize?
+    {
+        e->warning("Avoiding boundscheck for $");
+        return atEnd;
+    }
+    else if (e->op == TOKdiv)
+    {
+        DivExp* div = (DivExp*)e;
+        if (div->e1->op == TOKvar && this->isOpDollar((VarExp*)div->e1) && // TODO functionize?
+            div->e2->op == TOKint64)
+        {
+            *q = div->e2->toInteger();
+            if (*q >= 1)
+            {
+                e->warning("Avoiding boundscheck for $/%ld", *q);
```

Comment:
I assume you're planning to take these out, but don't use warnings for this if you want the autotester to pass.  Just use printf.